### PR TITLE
fix: keepValues true for react-hook-form

### PR DESCRIPTION
**Pull request by omeraplak**
Please provide enough information so that others can review your pull request:

Fixed resetting values on step changes

### Closing issues

- #3290

### Self Check before Merge

Please check all items below before review.

-   [x] Corresponding issues are created/updated or not needed
-   [x] Docs are updated/provided or not needed
-   [x] Examples are updated/provided or not needed
-   [x] TypeScript definitions are updated/provided or not needed
-   [x] Tests are updated/provided or not needed
-   [x] Changesets are provided or not needed

### CODE REVIEW

**Review by aliemir — APPROVED** [review]

File: packages/react-hook-form/src/useStepsForm/index.ts [hunk 1]
```diff
@@ -81,7 +81,12 @@ export const useStepsForm = <
                 }
             });
 
-            reset(fields as any, { keepDirty: true });
+            console.log("rest");
```
Comment (alicanerdurmaz) [inline packages/react-hook-form/src/useStepsForm/index.ts:84]:
```suggestion
```

**Review by alicanerdurmaz — APPROVED** [review]

File: packages/react-hook-form/src/useStepsForm/index.ts (hunk 1 above)
Comment (aliemir) [inline packages/react-hook-form/src/useStepsForm/index.ts:84]:
```suggestion
```

🤣

Comment (omeraplak) [reply packages/react-hook-form/src/useStepsForm/index.ts]: 🤣

Comment (omeraplak) [reply packages/react-hook-form/src/useStepsForm/index.ts]: 🚀

**Pull request merged**